src/app/form-page/form-page.component.html

```diff
@@ -0,0 +1,456 @@
+<div id="main">
+  <div id="header">
+    <div>
+      <img src="/assets/images/logo.png" id="logo"/>
+    </div>
+    <div>
+      <h1>IT Academija</h1>
+    </div>
+  </div>
+
+  <div id="form-container">
+    <h2 id="form-title">Registracijos forma</h2>
+    <form [formGroup]="applicationForm" (ngSubmit)="onSubmit()">
+      <div class="form-group" [class.has-error]="fname.invalid">
+        <label for="fname">Vardas:</label>
+        <input
+          type="text"
+          class="form-control"
+          id="fname"
+          formControlName="fname"
+        />
+        <div
+          [hidden]="fname.valid || fname.untouched"
```

Comment:
I'd personally just use *ngIf instead of [hidden] in all of the cases in this file

Reply:
Hidden  is woriking better here: using ngIf leaves empty red rectangle under an input element, while hidden just works fine.